portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,35 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.util.ArrayList;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+//Premade list of comments to pass from server
+  private ArrayList<String> comments;
+  private Gson gson = new Gson();
+
+//init() is called once when servlet is created
+  @Override
+  public void init() {
+      comments = new ArrayList<>();
+      comments.add("Pineapple on pizza is great");
+      comments.add("Dogs are overrated");
+      comments.add("Alpacas are awesome");
+  }
+
+//Get the list of comments from server as a Json
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Budi!</h1>");
+    String json = convertToJson(comments);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  private String convertToJson(ArrayList<String> comments) {
+      String json = gson.toJson(comments);
+      return json;
```

Comment:
This method is not necessary as it only does one instruction.
I think you can just inline it in doGet